test/StrStrTest.java

```diff
@@ -0,0 +1,41 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@RunWith(Parameterized.class)
+public class StrStrTest {
+    String haystack;
+    String needle;
+    int expectedResult;
+    String description;
+
+    public StrStrTest(String haystack, String needle,
+                      int expectedResult, String description) {
+        this.haystack = haystack;
+        this.needle = needle;
+        this.expectedResult = expectedResult;
+        this.description = description;
+    }
+
+    @Parameterized.Parameters
+    public static Collection numsAndTargets() {
+        return Arrays.asList(new Object[][] {
+                { "hello" , "ll", 2, "Substring is present" },
```

Comment:
Not enough test cases. 
Consider the following:
* strings match at the beginning 
* strings match at the end
* string are the same (full length match)
* haystack is substring of needle

Reply:
These tests were added and all of them have succeed, but tests still fail for empty haystack or needle, so could we please discuss this matter?